components/LspHero.tsx

```diff
@@ -63,7 +69,12 @@ export const LspHero: React.FC<Props> = ({
           </Button>
         </CardHead>
         {isConnected && <Account id="walletAccount">{account}</Account>}
-        {isConnected && (
+        {isConnected && !isUserConnectedToContractChain && (
```

Comment:
nothing wrong with the condition here, but it could get a bit hard to follow with multiple of those in the JSX, we could define a variable that tells us exactly what the issue is like `hasToChangeChain` and it could be defined in the component body above, so it does not clutter the JSX